### DISCUSSION TITLE
Trying to run database in our cluster is not working, But it is working in local

### DISCUSSION THREAD

**Discussion by nikhilreddy-pottanig** [milvus-io/milvus#35378] [Q&A and General discussion]
FROM milvusdb/milvus:latest as milvus_base

# Install curl for healthcheck
RUN apt-get update && apt-get install -y curl && rm -rf /var/lib/apt/lists/*

# Set environment variables
ENV ETCD_USE_EMBED=true
ENV COMMON_STORAGETYPE=local
ENV MILVUS_STORAGE_PATH=/var/lib/milvus

# Create directory for persistent storage
RUN mkdir -p /var/lib/milvus

# Expose necessary ports
EXPOSE 19530 9091 2379

# # Run Milvus standalone
CMD ["milvus", "run", "standalone"]

**Comment by yhmo** [comment]
Seems is not a standard milvus deployment.

**Reply by nikhilreddy-pottanig** [reply]
Can't I directly install Milvus and start database server without using docker

**Comment by nikhilreddy-pottanig** [comment]
Tired running this way commands to install Milvus wasn't working

RUN apt-get update && apt-get install -y \
    software-properties-common \
    && add-apt-repository ppa:milvusdb/milvus \
    && apt-get update \
    && apt-get install -y milvus \

**Reply by yhmo** [reply]
Milvus apt installation is supported in v2.3 but is deprecated in v2.4.
This doc is about apt installation with v2.3:
https://milvus.io/docs/v2.3.x/install_standalone-aptyum.md

**Reply by nikhilreddy-pottanig** [reply]
# Use Python base image with a slim version
FROM python:3.11-slim as gradio_ui

# Install system dependencies
RUN apt-get update && apt-get install -y \
    wget \
    gnupg \
    lsb-release \
    libaio1 \
    libgfortran5 \
    && rm -rf /var/lib/apt/lists/*

# Install Milvus
RUN wget https://github.com/milvus-io/milvus/releases/download/v2.3.16/milvus_2.3.16-1_amd64.deb \
    && apt-get update \
    && dpkg -i milvus_2.3.16-1_amd64.deb \
    && apt-get -f install -y \
    && rm milvus_2.3.16-1_amd64.deb

# Start Milvus service
CMD ["milvus", "run", "standalone"]

Is this the right way to use it?
Can't make above code working, getting connection error about etcd!
Can you help me out with what's the best way to start milvus without using milvus docker file

**Reply by yhmo** [reply]
I just follow the doc with these steps:
```
wget https://github.com/milvus-io/milvus/releases/download/v2.3.16/milvus_2.3.16-1_amd64.deb
sudo apt-get update
sudo dpkg -i milvus_2.3.16-1_amd64.deb
sudo apt-get -f install
sudo systemctl restart milvus
```
![Screenshot from 2024-08-13 12-31-52](https://github.com/user-attachments/assets/5889e534-40f1-4601-bf91-49d5ff5f2091)


I can see the milvus server is running:
![Screenshot from 2024-08-13 12-34-09](https://github.com/user-attachments/assets/16b1cd32-3dee-4f16-a055-d41fbc22ef90)

And I use pymilvus to test, it works well.

**Reply by nikhilreddy-pottanig** [reply]
# Use an official Ubuntu base image
FROM ubuntu:20.04 as milvus_base

# Set environment variables to ensure non-interactive installation
ENV DEBIAN_FRONTEND=noninteractive

# Update and install necessary packages
RUN apt-get update && apt-get install -y \
    wget \
    systemctl \
    && rm -rf /var/lib/apt/lists/*

# Download the Milvus .deb package
RUN wget https://github.com/milvus-io/milvus/releases/download/v2.3.16/milvus_2.3.16-1_amd64.deb

# Install the Milvus package
RUN apt-get update && dpkg -i milvus_2.3.16-1_amd64.deb || apt-get -f install -y

# # Restart the Milvus service
# RUN systemctl restart milvus

# Expose Milvus port
EXPOSE 19530

# Run Milvus in the foreground
CMD ["systemctl", "start", "milvus"]


Trying to run above docker content:-
Getting below errors
ERROR:systemctl:Unit milvus.service not for --user mode
ERROR:systemctl:can not open milvus.service log: /tmp/run-toolkit/log/journal/milvus.service.log
	[Errno 2] No such file or directory: '/tmp/run-toolkit/log/journal/milvus.service.log'

**Comment by nikhilreddy-pottanig** [comment]
<img width="1018" alt="Screenshot 2024-08-13 at 8 28 50 PM" src="https://github.com/user-attachments/assets/4d024a7d-44bd-4845-a313-3eb5e8229b2b">
When trying to use shell in the above container.
When typing milvus nothing is happening

**Reply by nikhilreddy-pottanig** [reply]
And I see only 2.3.16 has deb file what about 2.4?
Which mainly have the features I need and my cluster doesn't support Docker compose kinda Services

**Reply by xiaofan-luan** [reply]
we don't support deb file for 2.4 or later, due to we don't have enough people to manage it. 
Using docker/K8s is what we recommneded to do in production
